### PR TITLE
Add Row to corresponding group

### PR DESCRIPTION
**Pull request by Paraplegix**
Also added a quick fix to the double add row bug #193  in case self.data and self.activeData have the same reference

### CODE REVIEW

File: tabulator.js [hunk 1]
```diff
@@ -1260,15 +1260,32 @@
...
+				self.table.append(group);
+				self._renderGroupHeader(group);
+			}
+			//set the place to append the row to the corresponding group.
+			appendPlace = group;
+		}
 		//append to top or bottom of table based on preference
 		if(top == "top"){
-			self.activeData.unshift(item);
+			if (self.activeData != self.data)
```
Comment (olifolkerd) [inline tabulator.js]: Please use !== for object reference comparison

Comment (olifolkerd) [inline tabulator.js]: Please make sure all if statements use curly braces, it is better for code maintainability longer term.

File: tabulator.js [hunk 2]
```diff
@@ -1260,15 +1260,32 @@
...
+				self.activeData.unshift(item);
 			self.data.unshift(item);
-			self.table.prepend(row);
+			appendPlace.prepend(row);
 		}else{
-			self.activeData.push(item);
+			if (self.activeData != self.data)
```
Comment (olifolkerd) [inline tabulator.js]: Please use !== for object reference comparison

Comment (olifolkerd) [inline tabulator.js]: Please make sure all if statements use curly braces, it is better for code maintainability longer term.

File: tabulator.js [hunk 3]
```diff
@@ -1260,15 +1260,32 @@
 
 		var top = typeof top == "undefined" ? self.options.addRowPos : (top === true || top === "top" ? "top" : "bottom");
 
+		// initial place to append the row is the table
+		var appendPlace = self.table;
```
Comment (olifolkerd) [inline tabulator.js]: please change ***appendPlace*** variable name to ***newRowPosition*** for clarity

**Comment by Paraplegix** [comment]
Changes are made ! 
It should fit in with the coding style of the rest of Tabulator

**Pull request merged**